### ISSUE TITLE
`df.to_csv` drops index column from CSV entirely

### ISSUE THREAD

**Issue by RehanSD** [modin-project/modin#4014]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Modin version** (`modin.__version__`):
- **Python version**:
- **Code we can use to reproduce**:
```python
import modin.pandas as pd
cols = 100
arows = 20000
brows = 20000
keyrange = 100
a = pd.DataFrame(np.vstack([np.random.choice(keyrange,size=(arows)),np.random.normal(size=(cols,arows))]).transpose(),columns = ['key'] + ['avalue' + str(i) for i in range(1,1+cols)]).set_index('key')
b = pd.DataFrame(np.vstack([np.random.choice(keyrange,size=(brows)),np.random.normal(size=(cols,brows))]).transpose(),columns = ['key'] + ['bvalue' + str(i) for i in range(1,1+cols)]).set_index('key')
a.to_csv("a-big.csv")
b.to_csv("b-big.csv")
pd.read_csv("a-big.csv") # Has no key column
pd.read_csv("a-big.csv").set_index("key") # Errors
```
When using modin to write a Dataframe with a custom index to a csv file, the index is dropped entirely, and a range index is used when reading the Dataframe. Additionally, this may have been a transient error, but when I wrote the df using pandas, and then read it using modin, and called `set_index('key')` where `'key'` was the original index, the output displayed correctly, but I got the following `KeyError`:
```python
2022-01-20 13:23:49,267	ERROR worker.py:79 -- Unhandled error (suppress with RAY_IGNORE_UNHANDLED_ERRORS=1): ray::deploy_ray_func() (pid=78818, ip=127.0.0.1)
  File "/Users/rehandurrani/Documents/modin/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/axis_partition.py", line 211, in deploy_ray_func
    result = func(*args)
  File "/Users/rehandurrani/Documents/modin/modin/core/dataframe/pandas/partitioning/axis_partition.py", line 160, in deploy_axis_func
    result = func(dataframe, **kwargs)
  File "/Users/rehandurrani/Documents/modin/modin/core/dataframe/pandas/dataframe/dataframe.py", line 1310, in _tree_reduce_func
    series_result = func(df, *args, **kwargs)
  File "/Users/rehandurrani/Documents/modin/modin/core/storage_formats/pandas/query_compiler.py", line 441, in map_func
    return pandas.merge(left, right, **kwargs)
  File "/Users/rehandurrani/.miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/reshape/merge.py", line 106, in merge
    op = _MergeOperation(
  File "/Users/rehandurrani/.miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/reshape/merge.py", line 699, in __init__
    ) = self._get_merge_keys()
  File "/Users/rehandurrani/.miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/reshape/merge.py", line 1096, in _get_merge_keys
    right_keys.append(right._get_label_or_level_values(rk))
  File "/Users/rehandurrani/.miniconda3/envs/modin/lib/python3.8/site-packages/pandas/core/generic.py", line 1779, in _get_label_or_level_values
    raise KeyError(key)
KeyError: 'key'
```

The code for the replication of this bug was borrowed from marsupialtail's [question](https://discuss.modin.org/t/modin-is-slower-than-pandas/255) on the [modin discourse](discuss.modin.org).

**Comment by anmyachev** [comment]
Same problem reproduced for Dask engine, so the problem is not in the distributed version of `to_csv` that implemented for Ray.

**Comment by anmyachev** [comment]
The problem is not reproducible for 0.12.1 release, hence appeared later.

**Comment by anmyachev** [comment]
Found a commit that introduces the regression -https://github.com/modin-project/modin/commit/3c740dbfcdd69ddc3ab45a42be996e5c61104342.

@naren-ponder can you look at the problem?

**Comment by devin-petersohn** [comment]
I have put a fix in #4025 .

**Comment by RehanSD** [comment]
Reviewing #4025
